gadget/update.go

```diff
@@ -208,6 +223,22 @@ func EnsureLayoutCompatibility(gadgetLayout *LaidOutVolume, diskLayout *OnDiskVo
 		return false, reasonAbsent
 	}
 
+	onDiskContains := func(haystack []OnDiskStructure, needle LaidOutStructure) (bool, string) {
+		reasonAbsent := ""
+		for _, h := range haystack {
+			matches, reasonNotMatches := eq(h, needle)
+			if matches {
+				return true, ""
+			}
+			// this has the effect of only returning the last non-empty reason
+			// string
+			if reasonNotMatches != "" {
+				reasonAbsent = reasonNotMatches
```

Comment:
Would it make sense to collect all of these reasons, and then when returning either return all of them, or keep the code as it currently is, but just add a `log.Noticef` where we print all of them? Just wondering if that whould help debugging.

Reply:
yeah this could be done in a more elegant manner, I will see what we can do

Reply:
Hmm so I did look at this again, and basically the only situation I could find where we end up having multiple reasonNotMatches is where we have multiple structures in the YAML with the same name. 

Since the first loop over diskLayout.Structure to find a structure in the YAML matching each on disk partition will return an error if it cannot find a YAML structure matching the on disk one, in this case the only way we could have multiple error reasons here is if the name of the on disk partition we have matches multiple structures in the YAML, and none of the YAML structures with the same name actually fit/match. This seems highly unlikely to happen in practice, and furthermore I think that sort of YAML with multiple names should not be allowed at all much earlier in the process rather than here.

So I think what we should do to clean this up is close that loophole where we allow YAML's with duplicated names, and then we can leave a comment in this function that we don't need to handle the case of multiple errors because it's impossible without there being multiple YAML structures with the same name. I will leave a TODO about doing that work so that this PR can land quickly and handle in a followup actually closing the loophole.

Reply:
Oh but I guess the name check does not work on MBR so that's partially false :-/ In any case I would rather handle this in a followup, I will leave a TODO in any case

Reply:
and again I am proven wrong, we do actually validate that there are no duplicated names in the yaml, for DOS/MBR or GPT disks, so we actually don't need to worry about this at all like I originally though

Reply:
anyways there is now a TODO about this